static/js/server_events_dispatch.js

```diff
@@ -202,7 +202,6 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
             _.each(event.streams, function (stream) {
                 var sub = stream_data.get_sub_by_id(stream.stream_id);
                 stream_data.update_calculated_fields(sub);
-                subs.add_sub_to_table(sub);
```

Comment:
This should not be deleted from here. This adds the stream regardless of subscribed or not, to settings when stream is created. If we remove this from here, user will not have stream listed in settings until user is not subscribed to stream or do hard reload.